server/filestore.go

```diff
@@ -8111,13 +8131,20 @@ func (mb *msgBlock) dirtyCloseWithRemove(remove bool) {
 		// Clear any tracking by subject if we are removing.
 		mb.fss = nil
 		if mb.mfn != _EMPTY_ {
-			os.Remove(mb.mfn)
+			err := os.Remove(mb.mfn)
+			if err != nil && os.IsPermission(err){
+				return errFileSystemPermissionDenied
```

Comment:
Is there a reason to do this rather than just check `os.IsPermission` further up the callstack?

Reply:
It may even make sense to define a new error-detect function for this case, similar to how we already have this in `store.go` for out-of-space:
```
func isOutOfSpaceErr(err error) bool {
	return err != nil && (strings.Contains(err.Error(), "no space left"))
}
```